entity-framework/core/querying/raw-sql.md

```diff
@@ -137,14 +140,9 @@ There are a few limitations to be aware of when using raw SQL queries:
 
 * The SQL query cannot contain related data. However, in many cases you can compose on top of the query using the `Include` operator to return related data (see [Including related data](#including-related-data)).
 
-* `SELECT` statements passed to this method should generally be composable: If EF Core needs to evaluate additional query operators on the server (for example, to translate LINQ operators applied after `FromSql`), the supplied SQL will be treated as a subquery. This means that the SQL passed should not contain any characters or options that are not valid on a subquery, such as:
+* `SELECT` statements passed to this method should generally be composable: If EF Core needs to evaluate additional query operators on the server (for example, to translate LINQ operators applied after `FromSql` methods), the supplied SQL will be treated as a subquery. This means that the SQL passed should not contain any characters or options that are not valid on a subquery, such as:
   * a trailing semicolon
   * On SQL Server, a trailing query-level hint (for example, `OPTION (HASH JOIN)`)
   * On SQL Server, an `ORDER BY` clause that is not accompanied of `OFFSET 0` OR `TOP 100 PERCENT` in the `SELECT` clause
 
-* SQL statements other than `SELECT` are recognized automatically as non-composable. As a consequence, the full results of stored procedures are always returned to the client and any LINQ operators applied after `FromSql` are evaluated in-memory.
-
-> [!WARNING]  
-> **Always use parameterization for raw SQL queries:** In addition to validating user input, always use parameterization for any values used in a raw SQL query/command. APIs that accept a raw SQL string such as `FromSql` and `ExecuteSqlCommand` allow values to be easily passed as parameters. Overloads of `FromSql` and `ExecuteSqlCommand` that accept FormattableString also allow using string interpolation syntax in a way that helps protect against SQL injection attacks. 
-> 
-> If you are using string concatenation or interpolation to dynamically build any part of the query string, or passing user input to statements or stored procedures that can execute those inputs as dynamic SQL, then you are responsible for validating any input to protect against SQL injection attacks.
+* SQL statements other than `SELECT` are recognized automatically as non-composable. As a consequence, the full results of stored procedures are always returned to the client and any LINQ operators applied after `FromSql` methods are evaluated in-memory.
```

Comment:
AFAIR, the plan was that in 3.0 we were going to remove any heuristics to try to distinguish composable from non-composable queries. The latter should lead to incorrect SQL now. Did this plan change? 

cc @smitpatel

Reply:
The warning here is an important security note for EF Core. Why remove it? I think we should consider moving it to the top of the page instead, after making any adjustments necessary to it.

Reply:
The plan did not changed. The statement is wrong.

Reply:
Sorry. Have replaced the composability comment with one specifically addressing SQL Server stored procedures and client evaluation.

Reply:
> The warning here is an important security note for EF Core. Why remove it? I think we should consider moving it to the top of the page instead, after making any adjustments necessary to it.

Have brought the warning back, updated it and moved it to the top of the "Passing Parameters" section.